src/lib.rs

```diff
@@ -52,6 +52,26 @@ mod tests {
         tokio::run(consumer);
     }
 
+    #[test]
+    fn one_sync_element_collected_yield() {
```

Comment:
I'm unsure why we're only adding more tests here instead of replacing all of the existing tests with the packet collection assertion. Why?

Also, when all of our tests pass/fail on assertions, the `collected_yield` suffix will be redundant.

Reply:
Yes, I think this is true. Can we clean this up later during the testing fixup. I was did not want to depreciate existing work until this PR is in.